pkg/querier/stats/stats_test.go

```diff
@@ -78,35 +95,40 @@ func TestStats_AddFetchedDataBytes(t *testing.T) {
 
 func TestStats_Merge(t *testing.T) {
 	t.Run("merge two stats objects", func(t *testing.T) {
-		stats1 := &Stats{}
+		stats1 := &QueryStats{}
 		stats1.AddWallTime(time.Millisecond)
 		stats1.AddFetchedSeries(50)
 		stats1.AddFetchedChunkBytes(42)
 		stats1.AddFetchedDataBytes(100)
+		stats1.AddExtraFields("a", "b")
+		stats1.AddExtraFields("a", "b")
 
-		stats2 := &Stats{}
+		stats2 := &QueryStats{}
 		stats2.AddWallTime(time.Second)
 		stats2.AddFetchedSeries(60)
 		stats2.AddFetchedChunkBytes(100)
 		stats2.AddFetchedDataBytes(101)
+		stats2.AddExtraFields("c", "d")
 
 		stats1.Merge(stats2)
 
 		assert.Equal(t, 1001*time.Millisecond, stats1.LoadWallTime())
 		assert.Equal(t, uint64(110), stats1.LoadFetchedSeries())
 		assert.Equal(t, uint64(142), stats1.LoadFetchedChunkBytes())
 		assert.Equal(t, uint64(201), stats1.LoadFetchedDataBytes())
+		assert.ElementsMatch(t, []interface{}{"a", "b", "d", "c"}, stats1.LoadExtraFields())
```

Comment:
Why the result id "d", "c" rather than "c", "d"? Shall we sort the slice?

Reply:
This test is weird...

I just created a better test!